File: dvdrental/src/main/webapp/app/model/menu/Root.js
Ext.define('Sanjay.model.menu.Root', {
	extend : 'Ext.data.Model',
	uses : [ 'Sanjay.model.menu.Item' ],
	idProperty : 'id',
	fields : [ {
		name : 'text'
	}, {
		name : 'iconCls'
	}, {
		name : 'id'

	} ],
	hasMany : {
		model : 'Sanjay.model.menu.Item',
		foreignKey : 'parent_id',
		name : 'items'
	}
});

File: dvdrental/src/main/java/com/sanjay/dvdrental/model/Root.java
package com.sanjay.dvdrental.model;

import java.util.LinkedList;
import java.util.List;

public class Root {
	List<Menu> items = new LinkedList<Menu>();

	public List<Menu> getItems() {
		return items;
	}

	public void setItems(List<Menu> items) {
		this.items = items;
	}
}
